incognia.go

```diff
@@ -437,3 +445,32 @@ func isValidFeedbackType(feedbackType FeedbackType) bool {
 		return false
 	}
 }
+
+func getDefaultHeaders() map[string]string {
+	headers := make(map[string]string)
+	headers[contentTypeKey] = "application/json"
+	headers[userAgentKey] = generateUserAgentString()
+	return headers
+}
+
+func generateUserAgentString() string {
+	osName := runtime.GOOS
+	osArch := runtime.GOARCH
+
+	var uts syscall.Utsname
+	syscall.Uname(&uts)
```

Comment:
The syscall package is OS specific and the uname syscall is Linux specific. By adding this here you are breaking compatibility with macOS and Windows.

You should either use a OS independent function to get this information or implement this also for macOS and Windows, falling back to an empty string on other OSes.

Reply:
hmm, great call